### PR TITLE
feat(FN-974): Notify email for incoming report

### PR DESCRIPTION
**Pull request by francescastocco**
## Introduction
We need to notify PDC Inputters when the banks have submitted a utilisation report.

## Resolution
- Created endpoint in portal-api, which is called when user clicks `Confirm and Send` button on the utilisation report upload pages
- Used `sendEmail` function to use Notify to send email
- Added new email template to constants file
- Added API tests for role & auth validation

## To Do
Create an environment variable called `PDC_INPUTTERS_EMAIL_RECIPIENT`

### CODE REVIEW

File: portal-api/src/v1/controllers/utilisation-report-service/utilisation-report-upload.controller.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  );
+};
+
+const uploadReportAndSendNotification = async (req, res) => {
+  const { bankName, month, year } = req.body;
+  try {
+    // TODO FN-1103 save file
+    sendEmailToPdcInputtersEmail(bankName, month, year);
+    // TODO FN-1103 change what is sent in the response
+    res.status(200).send(month);
```
[the remark this answers is not in the document]
Comment (ChristianMcc) [reply portal-api/src/v1/controllers/utilisation-report-service/utilisation-report-upload.controller.js:23]: Maybe just return true or something here leaving in the comment to make the code scanning happy for now (know it's only temporary code!)

**Review by ChristianMcc — CHANGES_REQUESTED** [review]
LGTM! Small env variable related changes that might help you :D

File: docker-compose.dev.yml [hunk 2]
```diff
@@ -172,6 +172,7 @@ services:
       - TZ=Europe/London
       - NODE_ENV=${NODE_ENV}
       - RATE_LIMIT_THRESHOLD
+      - PDC_INPUTTERS_EMAIL_RECIPIENT=francesca.stocco@ukexportfinance.gov.uk
```
[the remark this answers is not in the document]
Comment (francescastocco) [reply docker-compose.dev.yml:175]: Oooh yes, that would be bad

File: portal-api/src/constants/email-template-ids.js [hunk 3]
```diff
@@ -3,11 +3,15 @@ const BLOCKED = '82506983-cb85-4f33-b962-922b850be7ac';
 const UNBLOCKED = '44959d08-6389-4f27-a6be-2faae8bea711';
 const PASSWORD_UPDATE = '41235821-7e52-4d63-a773-fa147362c5f0';
 const UPDATE_STATUS = '718beb52-474e-4f34-a8d7-ab0e48cdffce';
+const NEW_UTILISATION_REPORT = 'ba833a42-bad8-4f96-8d66-42bf1ce116f1';
+const PASSWORD_RESET = '6935e539-1a0c-4eca-a6f3-f239402c0987';
```
[the remark this answers is not in the document]
Comment (francescastocco) [reply portal-api/src/constants/email-template-ids.js:7]: Ooh yes, I refactored the [password reset file](https://github.com/UK-Export-Finance/dtfs2/pull/2161/files#diff-873955b5374628a0d2e97c56a34970251cb10cdd40a24fbbc8f0bc7a169adeb6) as I thought it was better to use a constant, is this ok or should I have left it? I did actually inadvertently test it worked when doing a password reset but doubt we'd regression test this?

**Review by ChristianMcc — APPROVED** [review]
Nice :)

**Review by ChristianMcc — APPROVED** [review]

**Pull request merged**